### PR TITLE
Task/upgrade web dependencies and fix readme

### PR DESCRIPTION
**Pull request by jmetev1**
I did minor package upgrades. I skipped @types/luxon. there's whole discussion here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64995/files#diff-0e8e2c47dc4c85e1be2e301151ad8cb7b73533189f48903cddc50b5c113d38e7R288 about repercussions. i figured i would punt on that. but i can update if you want. that's what was breaking https://github.com/bcgov/wps/pull/2930

i also thought i would be able to delete web/dockerfile bc i was under the impression that it was not being used. but that broke docker compose too. although i couldn't do docker compose either way due to output of The Poetry configuration is invalid: #0 0.704     - Additional properties are not allowed ('group' was unexpected)

I also tried readme instructions for dev container but their is no: select docker-compose.vscode.yml

also yarn test fails locally but seems to be passing in ci so 🤷

image
closes https://github.com/bcgov/wps/issues/2931

also all of the following files are unused. want them removed?
src/features/fba/components/FormalFBATable.tsx
src/features/fba/components/map/FBATooltip.tsx
src/features/fbaCalculator/data/data.ts
src/features/fireWeather/components/maps/LayerSwitch.tsx
src/features/moreCast2/constants.ts
# Test Links:
[Landing Page](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2935.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Wildfire Predictive Services to support decision making in prevention, preparedn
 
 1. Create `.env` file in `web` using `web/.env.example` as a sample.
 2. Create `.env.docker` file in `api/app` using `api/app/.env.example` as a sample.
-3. Run `docker-compose build` and then `docker-compose up`
+3. Run `docker compose build` and then `docker compose up`
```
Comment (jmetev1) [inline README.md:22]: web/readme has docker compose without the - and it's newer so i figured it was correct and changed ths one

File: web/package.json [hunk 2]
```diff
@@ -16,7 +16,6 @@
     "@emotion/react": "^11.8.2",
     "@emotion/styled": "^11.8.1",
     "@mui/icons-material": "^5.5.1",
-    "@mui/lab": "^5.0.0-alpha.75",
```
Comment (jmetev1) [inline web/package.json:19]: not used anymore

File: web/package.json [hunk 3]
```diff
@@ -26,7 +25,6 @@
     "@types/leaflet": "^1.7.0",
     "@types/lodash": "^4.14.173",
     "@types/luxon": "^3.0.2",
-    "@types/ol": "^6.4.2",
```
Comment (jmetev1) [inline web/package.json:29]: https://www.npmjs.com/package/@types/ol

File: web/package.json [hunk 4]
```diff
@@ -92,7 +90,6 @@
   "devDependencies": {
     "@cypress/code-coverage": "^3.10.0",
     "@cypress/instrument-cra": "^1.4.0",
-    "@testing-library/dom": "^9.0.0",
```
Comment (jmetev1) [inline web/package.json:95]: testing-library/react already requires testing-library/dom 8.0. i don't think there was any reason to separately install 9.0

**Comment by jmetev1** [comment]
ok i remade this from a non forked branch. that's what you wanted @conbrad ?

**Comment by conbrad** [comment]
Looks like we have 1 failing cypress test with these changes: https://github.com/bcgov/wps/actions/runs/5261086616/jobs/9508804076?pr=2935

**Comment by jmetev1** [comment]
i figured out that it's this upgrade: `@mui/material | ⬆️ | 5.11.14...5.13.5` that broke it. but i'm not sure why. before my changes, there was this happening which maybe seems undesirable. did you expect that a request is  failing during the tests? i guess it's one that's not getting intercepted and maybe doesn't matter.
image

so going forward i could
1. skip this upgrade so we could merge this.
2. spend the time to figure out why it's happening and what to do about it.
what do you prefer?

**Comment by conbrad** [comment]
> i figured out that it's this upgrade: `@mui/material | ⬆️ | 5.11.14...5.13.5` that broke it. but i'm not sure why. before my changes, there was this happening which maybe seems undesirable. did you expect that a request is failing during the tests? i guess it's one that's not getting intercepted and maybe doesn't matter. image
> 
> so going forward i could
> 
> 1. skip this upgrade so we could merge this.
> 2. spend the time to figure out why it's happening and what to do about it.
>    what do you prefer?

Interesting! Thanks for looking deeper into the issue. I'm fine with either approach; we could pull out an issue specifically for the `@mui/material` package to get the other updates up to date in the meantime.

File: web/package.json [hunk 5]
```diff
@@ -16,8 +16,7 @@
     "@emotion/react": "^11.8.2",
     "@emotion/styled": "^11.8.1",
     "@mui/icons-material": "^5.5.1",
-    "@mui/lab": "^5.0.0-alpha.75",
-    "@mui/material": "^5.5.3",
+    "@mui/material": "5.12.2",
```
Comment (jmetev1) [inline web/package.json:19]: i locked this version bc 5.13.3 breaks a test i think due to a datepicker. could not figure out why.

**Comment by jmetev1** [comment]
i tried to debug for a while but gave up. i figured out the exact breaking version and read changelog and all but i'm not sure what's happening. part of the problem is that i haven't figured out how to login. so i'm doing everything by just watching cypress run. when i run the app and click on `hfi calculator > get started` brings me here. 
image

**Comment by conbrad** [comment]
> i tried to debug for a while but gave up. i figured out the exact breaking version and read changelog and all but i'm not sure what's happening. part of the problem is that i haven't figured out how to login. so i'm doing everything by just watching cypress run. when i run the app and click on `hfi calculator > get started` brings me here. image

Ok good to know, this is what breaks for the `@mui/material` upgrade? It's interesting you see the SSO page since that should be bypassed by our Cypress tests. I can add this to the follow up ticket to upgrade that dependency.

**Comment by jmetev1** [comment]
sorry no i wasn't very clear. cypress does not encounter the sso page. but i do when i try to login. i was trying to go to hfi calculator to recreate the behavior. should i be able to access it?

**Comment by conbrad** [comment]
> sorry no i wasn't very clear. cypress does not encounter the sso page. but i do when i try to login. i was trying to go to hfi calculator on my local instance to recreate the behavior. should i be able to access it? i can get to http://localhost:3000/ but not http://localhost:3000/hfi-calculator

Ah I see that makes sense. Several pages are behind auth, and you're our first external contributor :)

I'll see about making a locally runnable configuration that doesn't require auth.

**Review by conbrad — APPROVED** [review]

**Comment by jmetev1** [comment]
cool cool. beyond the react and axios upgrades, what would you like done? i don't know python but could learn. but i do see 
https://github.com/bcgov/wps/issues/853 /idea in there of moving to vite

**Comment by conbrad** [comment]
> cool cool. beyond the react and axios upgrades, what would you like done? i don't know python but could learn. but i do see #853 /idea in there of moving to vite

You're totally welcome to take on any issue you'd like to do! Moving to `vite` is high on my priority list but having that play nice with cypress test instrumentation is tricky.

> i locked this version bc 5.13.3 breaks a test i think due to a datepicker. could not figure out why.
So it looks like our date picker does break with this change -- the dropdown doesn't render when clicked. I'll try to undo the upgrade to the material library, or bisect the breaking change if I have some time later.

**Comment by jmetev1** [comment]
> So it looks like our date picker does break with this change -- the dropdown doesn't render when clicked. I'll try to undo the upgrade to the material library, or bisect the breaking change if I have some time later.

so i locked it to "5.12.2" because that passed the test. are you saying that with 5.12.2 the datepicker is not working? i thought it was only breaking on 5.12.3

**Comment by conbrad** [comment]
> > So it looks like our date picker does break with this change -- the dropdown doesn't render when clicked. I'll try to undo the upgrade to the material library, or bisect the breaking change if I have some time later.
> 
> so i locked it to "5.12.2" because that passed the test. are you saying that with 5.12.2 the datepicker is not working? i thought it was only breaking on 5.12.3

Yep, looks to be the case. I don't see any errors in console showing up.

**Comment by conbrad** [comment]
> excellent. anything else i need to do here? Socket Security flagged one new package. but everything else is passing

Yeah so I'm just going to debug this date picker issue a bit this afternoon -- our dev deployment has the issue, but running it locally it does not. 

I looked into the Socket Security flag: https://socket.dev/npm/package/mz/issues/2.7.0
It's a dependency from `react-scripts` which is one reason I'd like to move to vite.

**Comment by conbrad** [comment]
> > excellent. anything else i need to do here? Socket Security flagged one new package. but everything else is passing
> 
> Yeah so I'm just going to debug this date picker issue a bit this afternoon -- our dev deployment has the issue, but running it locally it does not.

Ok so the issue was our dev deployment going to revert my last commit.

**Comment by conbrad** [comment]
@SocketSecurity ignore mz@2.7.0

**Comment by jmetev1** [comment]
i'm also trying to run the whole app so i can do stuff requiring the backend. i've been trying to follow instructions using docker compose in root readme. but getting poetry errors. am i following the right instructions to start with? i also tried the instructions in `Developing the application in a dev container, using vscode:` but there is no such thing as `docker-compose.vscode.yml`

**Pull request merged**